src/analysis/processing/qgsalgorithmaddtablefield.cpp

```diff
@@ -75,10 +76,18 @@ void QgsAddTableFieldAlgorithm::initParameters( const QVariantMap & )
 {
   addParameter( new QgsProcessingParameterString( QStringLiteral( "FIELD_NAME" ), QObject::tr( "Field name" ) ) );
   addParameter( new QgsProcessingParameterEnum( QStringLiteral( "FIELD_TYPE" ), QObject::tr( "Field type" ),
-                QStringList() << QObject::tr( "Integer" ) << QObject::tr( "Float" ) << QObject::tr( "String" )
-                << QObject::tr( "Boolean" ) << QObject::tr( "Date" ) << QObject::tr( "Time" )
-                << QObject::tr( "DateTime" ) << QObject::tr( "Binary" )
-                << QObject::tr( "StringList" ) << QObject::tr( "IntegerList" ) << QObject::tr( "DoubleList" ), false, 0 ) );
+                QStringList() << QgsVariantUtils::typeToDisplayString( QVariant::Int )
```

Comment:
@nyalldawson , unfortunately, the ordering of the items matters here, can't be changed until at least QGIS 4.0 because the alg. looks at the list index integer to determine file type.

In this case, it means the Integer Float and String have to stay 1st, 2nd, and 3rd. The rest is all new so doesn't matter.

Reply:
Also, there's the field calculator algorithm that'd need updating (I can do it separately if you don't want to, just pointing it out).

Reply:
@nirvn I don't think the order has changed here

Reply:
(Just realized I had the radio box set to approved -- the ordering does need to be addressed)

Reply:
Arg, ignore me, you are right, Real vs Double reference confused me. All good then.

Reply:
Field calculator strings done now too!